sdk/go/pulumi/rpc.go

```diff
@@ -91,10 +91,16 @@ func mapStructTypes(from, to reflect.Type) func(reflect.Value, int) (reflect.Str
 // * Cust4 because it is a child of a custom resource
 // * Comp2 because it is a non-remote component resoruce
 // * Comp3 and Cust5 because Comp3 is a child of a remote component resource
-func addDependency(ctx context.Context, deps urnSet, res Resource) error {
+func addDependency(ctx context.Context, deps urnSet, res, from Resource) error {
 	if _, custom := res.(CustomResource); !custom {
+		// If `res` is the same as `from`, exit early to avoid depending on
+		// children that haven't been registered yet.
+		if res == from {
```

Comment:
This checks for pointer identity, right? Is it correct to assume that pointer identity? _i.e._ Are we sure that the concrete type of `res` and `from` will have identical pointers?
I think if `from` is a deep copy of `res`, this condition will yield a false negative. The alternative would be to use a deep equals here.

Reply:
> This checks for pointer identity, right? Is it correct to assume that pointer identity? _i.e._ Are we sure that the concrete type of `res` and `from` will have identical pointers?

Yeah, this should be OK. You generally have the same pointer identity for resources in a program and elsewhere we compare these in similar ways (not deep):

https://github.com/pulumi/pulumi/blob/9e73c75cd63215fc7bba16321eefbc2576719eb8/sdk/go/pulumi/resource_set.go#L3-L7

https://github.com/pulumi/pulumi/blob/9e73c75cd63215fc7bba16321eefbc2576719eb8/sdk/go/pulumi/resource.go#L91-L99

Reply:
Thanks for the FYI! :)